### ISSUE TITLE
Interface: Fractional Scale

### ISSUE THREAD

**Issue by ghost** [OpenTTD/OpenTTD#6797]
I always double the size of the interface, but it would be nice to have more options available.

Ideas:
- more percentage options: 125%, 150%, 175%, etc.
- a horizontal bar to increase/decrease the size instead of a list, I've seen on other games and it's a intuitive way of changing

**Comment by PeterN** [comment]
Would be nice. Currently GUI scaling works based on zoom levels which is why only powers of two are possible.

**Comment by nielsmh** [comment]
Another upvote from me, on my 22 inch 1920x1080 display the default UI scale is a bit too small, but the double scale takes up too much space.

I think one way to achieve this as-is would be to "simply" draw new UI graphics in e.g. 125% or 150% scale, and perhaps use a True Type font in somewhat larger size. None of the sprite art will scale nicely in non-integer amounts anyway.

**Comment by andythenorth** [comment]
125% would suit me right down to the ground, but I don't know if the pixel icons will upscale nicely for arbitrary multipliers.

**Comment by nielsmh** [comment]
An experiment worth trying might be to just add some interior padding on icon-buttons to make them larger targets on screen, without actually scaling the icon at all? I think that might just work for 125%, together with a larger font.

**Comment by nielsmh** [comment]
Here's a quick and dirty test of how the main toolbar looks with a bit of padding added to all the buttons. I would certainly play with this.

![image](https://user-images.githubusercontent.com/1062071/40684445-aaed0d26-6391-11e8-9145-0f2d34e32a5e.png)

**Comment by ghost** [comment]
That's a good idea, some padding would be nice indeed, it looks more organized, I like it!

**Comment by J0anJosep** [comment]
@nielsmh I wrote the code for adding such padding, even providing the user a GUI for selecting the size for the buttons, the size for the height of elements on the lists and choosing the fonts and their sizes.
I also added an autosizing option, where the program automatically sets the scaling for GUI sprites and the sizes of the rest of the elements of the GUI.
The result is good enough: I can switch between screen resolutions at ease without worrying about font sizes and sprites. The only thing I miss is that 100%-200%-400% GUI sprites aren't enough for a smooth transition between all resolutions (from 640x400 to 4k resolution).
